tests/tasks/snowflake/test_snowflake.py

```diff
@@ -50,29 +91,121 @@ def test_execute_fetchall(self, monkeypatch):
         """
         Tests that the SnowflakeQuery Task calls the fetchall method on the
         cursor. This is to prevent future code edits from returning the cursor
-        object because that cursors are not pickleable.
+        object because cursors are not pickleable.
         """
-        snowflake_module_connect_method = MagicMock()
-        connection = MagicMock(spec=sf.SnowflakeConnection)
         cursor = MagicMock(spec=sf.DictCursor)
-
-        # link all the mocks together appropriately
-        snowflake_module_connect_method.return_value = connection
-        connection.cursor = cursor
+        connection = MagicMock(spec=sf.SnowflakeConnection, cursor=cursor)
+        snowflake_module_connect_method = MagicMock(return_value=connection)
+        snowflake_connector_module = MagicMock(connect=snowflake_module_connect_method)
 
         # setting fetchall return
         cursor.return_value.__enter__.return_value.execute.return_value.fetchall.return_value = [
             "TESTDB"
         ]
-        snowflake_connector_module = MagicMock(connect=snowflake_module_connect_method)
 
         monkeypatch.setattr(
             "prefect.tasks.snowflake.snowflake.sf", snowflake_connector_module
         )
 
-        query = "SHOW DATABASES"
         output = SnowflakeQuery(
-            account="test", user="test", password="test", query=query
+            account="test", user="test", password="test", query="SHOW DATABASES"
         ).run()
 
         assert output == ["TESTDB"]
+
+
+class TestSnowflakeQueriesFromFile:
+    def test_construction(self):
+        task = SnowflakeQueriesFromFile(
+            account="test", user="test", password="test", warehouse="test"
+        )
+        assert task.autocommit is None
+
+    def test_runtime_arguments(self, monkeypatch, tmpdir, sql_file):
+        cursor = MagicMock(spec=sf.DictCursor)
+        connection = MagicMock(spec=sf.SnowflakeConnection)
+        snowflake_module_connect_method = MagicMock(return_value=connection)
+        snowflake_connector_module = MagicMock(connect=snowflake_module_connect_method)
+
+        # link all the mocks together appropriately
+        connection.execute_string.return_value = [cursor]
+
+        # setting fetchall return
+        cursor.fetchall.return_value = "TESTDB"
+
+        monkeypatch.setattr(
+            "prefect.tasks.snowflake.snowflake.sf", snowflake_connector_module
+        )
+
+        # task needs to allow for runtime arguments
+        output = SnowflakeQueriesFromFile().run(
+            account="test", user="test", password="test", file_path=sql_file
+        )
+
+        # The result is a list because multiple queries are executed
+        assert output == ["TESTDB"]
+
+    def test_required_parameters(self):
```

Comment:
I think we want `SnowflakeQueriesFromFile` instead of `SnowflakeQuery` here right?